cylc/flow/config.py

```diff
@@ -1127,6 +1131,100 @@ def check_param_env_tmpls(self):
                 )
             )
 
+    def check_completion_expressions(self):
+        """Check any configured completion expressions.
+
+        * Ensure completion expressions are not used in Cylc 7 compat mode.
+        * Ensure completion expressions do not reference missing task outputs.
+        * Ensure completion expressions are valid and do not use forbidden
+          syntax.
+        * Ensure optional/required outputs are consistent between the
+          completion expression and the graph.
+
+        Raises:
+            WorkflowConfigError: If any of the above checks fail.
+
+        """
+        for name, rtcfg in self.cfg['runtime'].items():
+            expr = rtcfg.get('completion')
+            if not expr:
+                # no expression to validate
+                continue
+
+            if 'submit-failed' in expr:
+                raise WorkflowConfigError(
+                    f'Error in [runtime][{name}]completion:'
+                    f'\nUse "submit_failed" rather than "submit-failed"'
+                    ' in completion expressions.'
+                )
```

Comment:
An error will be raised if `-` chars appear in the expression as these nodes are not whitelisted.

I added this to provide a more helpful error message for the one built-in output which contains a hyphen character.

Reply:
I see but the error message is
```
Invalid expression: succeeded and x-y
"Sub" not permitted
```
which is fairly unclear IMO

Reply:
Which is why I added this message!

Reply:
Can we not just check if the string contains `-` and give an error saying use underscore instead?

Reply:
This example should error because subtraction is not an allowed operation:

```
succeeded - failed
```

This example should fail because they should replace `-` with `_`:

```
my-output
```

Reply:
IMO it's less confusing overall if we always error with "`-` invalid, use underscore instead", even if the user meant to do subtraction (whatever that would even mean to them)

Reply:
Ok, added a catch all for this.